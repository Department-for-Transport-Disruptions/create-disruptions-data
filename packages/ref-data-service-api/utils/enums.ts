export enum RefVehicleMode {
    bus = "bus",
    tram = "tram",
    coach = "coach",
    ferry = "ferry",
    underground = "underground",
    rail = "rail",
    metro = "metro",
    blank = "",
}
