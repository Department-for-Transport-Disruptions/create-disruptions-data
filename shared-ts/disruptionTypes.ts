import { z } from "zod";
import {
    consequenceOperatorsSchema,
    consequenceSchema,
    disruptionInfoSchema,
    disruptionSchema,
    networkConsequenceSchema,
    operatorConsequenceSchema,
    serviceSchema,
    servicesConsequenceSchema,
    stopSchema,
    stopsConsequenceSchema,
    validitySchema,
} from "./disruptionTypes.zod";

export type Disruption = z.infer<typeof disruptionSchema>;
export type DisruptionInfo = z.infer<typeof disruptionInfoSchema>;

export type Validity = z.infer<typeof validitySchema>;

export type Consequence = z.infer<typeof consequenceSchema>;
export type ConsequenceOperators = z.infer<typeof consequenceOperatorsSchema>;
export type Stop = z.infer<typeof stopSchema>;
export type Service = z.infer<typeof serviceSchema>;

export type NetworkConsequence = z.infer<typeof networkConsequenceSchema>;
export type OperatorConsequence = z.infer<typeof operatorConsequenceSchema>;
export type StopsConsequence = z.infer<typeof stopsConsequenceSchema>;
export type ServicesConsequence = z.infer<typeof servicesConsequenceSchema>;

export const routesSchema = z.object({
    inbound: z.array(stopSchema.partial()),
    outbound: z.array(stopSchema.partial()),
});

export type Routes = z.infer<typeof routesSchema>;
