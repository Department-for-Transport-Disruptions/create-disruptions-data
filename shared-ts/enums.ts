export enum SourceType {
    directReport = "directReport",
    email = "email",
    phone = "phone",
    fax = "fax",
    post = "post",
    feed = "feed",
    radio = "radio",
    tv = "tv",
    web = "web",
    pager = "pager",
    text = "text",
    other = "other",
}

export enum Progress {
    draft = "draft",
    open = "open",
    published = "published",
    closing = "closing",
    closed = "closed",
    pendingApproval = "pendingApproval",
    editPendingApproval = "editPendingApproval",
    draftPendingApproval = "draftPendingApproval",
    rejected = "rejected",
}

export enum MiscellaneousReason {
    accident = "accident",
    securityAlert = "securityAlert",
    congestion = "congestion",
    roadClosed = "roadClosed",
    incident = "incident",
    routeDiversion = "routeDiversion",
    unknown = "unknown",
    vandalism = "vandalism",
    overcrowded = "overcrowded",
    operatorCeasedTrading = "operatorCeasedTrading",
    vegetation = "vegetation",
    roadworks = "roadworks",
    specialEvent = "specialEvent",
    insufficientDemand = "insufficientDemand",
}

export enum PersonnelReason {
    unknown = "unknown",
    staffSickness = "staffSickness",
    staffInjury = "staffInjury",
    contractorStaffInjury = "contractorStaffInjury",
    staffAbsence = "staffAbsence",
    staffInWrongPlace = "staffInWrongPlace",
    staffShortage = "staffShortage",
    industrialAction = "industrialAction",
    unofficialIndustrialAction = "unofficialIndustrialAction",
    workToRule = "workToRule",
    undefinedPersonnelProblem = "undefinedPersonnelProblem",
}

export enum EquipmentReason {
    unknown = "unknown",
    pointsFailure = "pointsFailure",
    signalProblem = "signalProblem",
    trainWarningSystemProblem = "trainWarningSystemProblem",
    trackCircuitProblem = "trackCircuitProblem",
    signalFailure = "signalFailure",
    derailment = "derailment",
    engineFailure = "engineFailure",
    tractionFailure = "tractionFailure",
    breakDown = "breakDown",
    technicalProblem = "technicalProblem",
    brokenRail = "brokenRail",
    poorRailConditions = "poorRailConditions",
    wheelImpactLoad = "wheelImpactLoad",
    lackOfOperationalStock = "lackOfOperationalStock",
    defectiveFireAlarmEquipment = "defectiveFireAlarmEquipment",
    defectivePlatformEdgeDoors = "defectivePlatformEdgeDoors",
    defectiveCctv = "defectiveCctv",
    defectivePublicAnnouncementSystem = "defectivePublicAnnouncementSystem",
    ticketingSystemNotAvailable = "ticketingSystemNotAvailable",
    repairWork = "repairWork",
    constructionWork = "constructionWork",
    maintenanceWork = "maintenanceWork",
    emergencyEngineeringWork = "emergencyEngineeringWork",
    lateFinishToEngineeringWork = "lateFinishToEngineeringWork",
    powerProblem = "powerProblem",
    fuelProblem = "fuelProblem",
    swingBridgeFailure = "swingBridgeFailure",
    escalatorFailure = "escalatorFailure",
    liftFailure = "liftFailure",
    gangwayProblem = "gangwayProblem",
    closedForMaintenance = "closedForMaintenance",
    fuelShortage = "fuelShortage",
    deicingWork = "deicingWork",
    wheelProblem = "wheelProblem",
    luggageCarouselProblem = "luggageCarouselProblem",
    undefinedEquipmentProblem = "undefinedEquipmentProblem",
}

export enum EnvironmentReason {
    unknown = "unknown",
    fog = "fog",
    roughSea = "roughSea",
    heavySnowFall = "heavySnowFall",
    driftingSnow = "driftingSnow",
    blizzardConditions = "blizzardConditions",
    heavyRain = "heavyRain",
    strongWinds = "strongWinds",
    stormConditions = "stormConditions",
    stormDamage = "stormDamage",
    tidalRestrictions = "tidalRestrictions",
    highTide = "highTide",
    lowTide = "lowTide",
    ice = "ice",
    frozen = "frozen",
    hail = "hail",
    sleet = "sleet",
    highTemperatures = "highTemperatures",
    flooding = "flooding",
    waterlogged = "waterlogged",
    lowWaterLevel = "lowWaterLevel",
    highWaterLevel = "highWaterLevel",
    fallenLeaves = "fallenLeaves",
    fallenTree = "fallenTree",
    landslide = "landslide",
    undefinedEnvironmentalProblem = "undefinedEnvironmentalProblem",
    lightningStrike = "lightningStrike",
    sewerOverflow = "sewerOverflow",
    grassFire = "grassFire",
}

export enum DayType {
    monday = "monday",
    tuesday = "tuesday",
    wednesday = "wednesday",
    thursday = "thursday",
    friday = "friday",
    saturday = "saturday",
    sunday = "sunday",
}

export enum VehicleMode {
    bus = "bus",
    tram = "tram",
    ferryService = "ferryService",
    rail = "rail",
    underground = "underground",
}

export enum Severity {
    unknown = "unknown",
    normal = "normal",
    verySlight = "verySlight",
    slight = "slight",
    severe = "severe",
    verySevere = "verySevere",
}

export enum UserGroups {
    systemAdmins = "system-admins",
    orgAdmins = "org-admins",
    orgPublishers = "org-publishers",
    orgStaff = "org-staff",
    operators = "operators",
}

export enum PublishStatus {
    draft = "DRAFT",
    published = "PUBLISHED",
    editing = "EDITING",
    pendingApproval = "PENDING_APPROVAL",
    rejected = "REJECTED",
    editPendingApproval = "EDIT_PENDING_APPROVAL",
    pendingAndEditing = "PENDING_EDITING",
}

export enum SocialMediaPostStatus {
    pending = "Pending",
    successful = "Successful",
    rejected = "Rejected",
}

export enum Datasource {
    tnds = "tnds",
    bods = "bods",
}

export enum Modes {
    bus = "bus",
    coach = "coach",
    tram = "tram",
    ferry = "ferry",
    rail = "rail",
    underground = "underground",
    metro = "metro",
}
