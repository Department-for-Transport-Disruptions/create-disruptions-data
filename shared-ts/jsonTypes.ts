
export enum SourceType {
    directReport,
    email,
    phone,
    ax,
    post,
    feed,
    radio,
    tv,
    web,
    pager,
    text,
    other,
}

export enum Progress {
    draft, 
    pendingApproval, 
    approvedDraft, 
    open, 
    closing, 
    closed, 
    rejected
}

export enum MiscellaneousReason {
    accident,
    securityAlert,
    congestion,
    roadWorks,
    roadClosed,
    bombExplosion,
    incident,
    routeDiversion,
    staffAbsence,
    signalProblem,
    maintenanceWork,
    constructionWork,
    fog,
    ice,
    heavyRain,
    waterLogged,
    unknown,
    nearMiss,
    safetyViolation,
    signalPassedAtDanger,
    stationOverrun,
    trainDoor,
    emergencyServicesCall,
    policeRequest,
    fireBrigadeSafetyChecks,
    unattendedBag,
    telephonedThreat,
    suspectVehicle,
    civilEmergency,
    airRaid,
    sabotage,
    bombAlert,
    attach,
    evacuation,
    terroristIncident,
    gunfireOnRoadway,
    explosion,
    explosionHazard,
    securityIncident,
    fire,
    linesideFire,
    vandalism,
    passengerAction,
    staffAssault,
    railwayCrime,
    fatality,
    personUnderTrain,
    personHitByTrain,
    personIllOnVehicle,
    emergencyServices,
    collision,
    overcrowded,
    insufficientDemand,
    lightingFailure,
    leaderBoardFailure,
    serviceIndicatorFailure,
    serviceFailure,
    operatorCeasedTrading,
    operatorSuspended,
    routeBlockage,
    personOnTheLine,
    vehicleOnTheLine,
    levelCrossingIncident,
    objectOnTheLine,
    fallenTreeOnTheLine,
    vegetation,
    trainStruckAnimal,
    trainStruckObject,
    animalOnTheLine,
    roadworks,
    specialEvent,
    march,
    procession,
    demonstration,
    publicDisturbance,
    filterBlockade,
    sightseersObstructingAccess,
    bridgeStrike,
    overheadObstruction,
    undefinedProblem,
    problemsAtBorderPost,
    problemsAtCustomsPost,
    problemsOnLocalRoad
}

export enum PersonnelReason {
    unknown,
    staffSickness,
    staffInjury,
    contractorStaffInjury,
    staffAbsence,
    staffInWrongPlace,
    staffShortage,
    industrialAction,
    unofficialIndustrialAction,
    workToRule,
    undefinedPersonnelProblem
}

export enum EquipmentReason {
    unknown,
    pointsProblem,
    pointsFailure,
    signalProblem,
    trainWarningSystemProblem,
    trackCircuitProblem,
    signalFailure,
    derailment,
    engineFailure,
    tractionFailure,
    breakDown,
    technicalProblem,
    brokenRail,
    poorRailConditions,
    wheelImpactLoad,
    lackOfOperationalStock,
    defectiveFireAlarmEquipment,
    defectivePlatformEdgeDoors,
    defectiveCctv,
    defectivePublicAnnouncementSystem,
    ticketingSystemNotAvailable,
    repairWork,
    constructionWork,
    maintenanceWork,
    emergencyEngineeringWork,
    lateFinishToEngineeringWork,
    powerProblem,
    fuelProblem,
    swingBridgeFailure,
    escalatorFailure,
    liftFailure,
    gangwayProblem,
    closedForMaintenance,
    fuelShortage,
    deicingWork,
    wheelProblem,
    luggageCarouselProblem,
    undefinedEquipmentProblem
}

export enum EnvironmentReason {
    unknown,
    fog,
    roughSea,
    heavySnowFall,
    driftingSnow,
    blizzardConditions,
    heavyRain,
    strongWinds,
    stormConditions,
    stormDamage,
    tidalRestrictions,
    highTide,
    lowTide,
    ice,
    frozen,
    hail,
    sleet,
    highTemperatures,
    flooding,
    waterlogged,
    lowWaterLevel,
    highWaterLevel,
    fallenLeaves,
    fallenTree,
    landslide,
    undefinedEnvironmentalProblem,
    lightningStrike,
    sewerOverflow,
    grassFire
}

export enum DayType {
    monday,
    tuesday,
    wednesday,
    thursday,
    friday,
    saturday,
    sunday
}

export type Source = {
    SourceType: SourceType;
    TimeOfCommunication: string;
};

export type Period = {
    StartTime: string;
    EndTime?: string;
};

export type InfoLink = {
    Uri: string;
}


export type Reference = {
    RelatedToRef: string[];
};

export type Repetitions = {
    DayType: DayType[];
}

export type InfoLinks = {
    InfoLink: InfoLink[];
  }

export type BasePtSituationElement = {
    CreationTime: string;
    ParticipantRef: string;
    SituationNumber: string;
    Version?: Number;
    References?: Reference;
    Source: Source;
    Progress: Progress;
    ValidityPeriod: Period[];
    Repetitions?: Repetitions;
    PublicationWindow: Period;
    Planned: boolean;
    Summary: string;
    Description: string;
    InfoLinks?: InfoLinks;
};

export type MiscReasonPtSituationElement = BasePtSituationElement & {
  MiscellaneousReason: MiscellaneousReason;
};

export type PersonnelReasonPtSituationElement = BasePtSituationElement & {
  PersonnelReason: PersonnelReason;
};

export type EquipmentReasonPtSituationElement = BasePtSituationElement & {
  EquipmentReason: EquipmentReason;
};

export type EnvironmentReasonPtSituationElement = BasePtSituationElement & {
  EnvironmentReason: EnvironmentReason;
};

export type PtSituationElement = MiscReasonPtSituationElement | PersonnelReasonPtSituationElement | EquipmentReasonPtSituationElement | EnvironmentReasonPtSituationElement;

export type SituationExchangeDelivery = {
    ResponseTimestamp: string;
    Status?: boolean;
    ShortestPossibleCycle?: string;
    Situations: PtSituationElement[];
}

export type ServiceDelivery = {
    ResponseTimestamp: string;
    ProducerRef: string;
    ResponseMessageIdentifier: string;
    SituationExchangeDelivery: SituationExchangeDelivery;
}