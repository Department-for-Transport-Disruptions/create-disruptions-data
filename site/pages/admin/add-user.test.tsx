import { UserGroups } from "@create-disruptions-data/shared-ts/enums";
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import AddUser, { AddUserPageProps } from './add-user.page';

const blankInputs: AddUserPageProps = {
    inputs: {},
    errors: [],
};

const withInputs: AddUserPageProps = {
    inputs: {
        givenName: "dummy",
        familyName: "user",
        email: "dummy.user@gmail.com",
        group: UserGroups.orgAdmins,
    },
    errors: [],
};

describe("AddUser", () => {
    it("should render correctly when there are no inputs", () => {
        const { asFragment } = render(<AddUser {...blankInputs} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly with inputs", () => {
        const { asFragment } = render(<AddUser {...withInputs} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
