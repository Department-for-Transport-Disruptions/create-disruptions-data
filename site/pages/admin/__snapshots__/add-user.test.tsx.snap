// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddUser > should render correctly when there are no inputs 1`] = `
<DocumentFragment>
  <div
    class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and 
        <a
          class="govuk-link"
          href="/feedback"
          id="feedback-link"
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>
  <div
    class="govuk-width-container"
  >
    <button
      class="govuk-back-link mb-7"
      type="button"
    >
      Back
    </button>
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/admin/add-user"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h1
              class="govuk-heading-xl"
            >
              Add new user
            </h1>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="given-name"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="given-name-input"
                >
                  First name
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="given-name-input"
                    maxlength="100"
                    name="givenName"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="family-name"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="family-name-input"
                >
                  Last name
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="family-name-input"
                    maxlength="100"
                    name="familyName"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="email"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="email-input"
                >
                  Email address
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="email-input"
                    maxlength="100"
                    name="email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Organisation
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
            <div
              class="govuk-form-group"
            >
              <fieldset
                class="govuk-fieldset"
                id="group"
              >
                <legend
                  class="govuk-fieldset__legend"
                >
                  <span
                    class="govuk-heading-s govuk-!-margin-bottom-0"
                  >
                    What account do they require?
                  </span>
                </legend>
                <div
                  class=""
                >
                  <div
                    class="govuk-radios"
                    data-module="govuk-radios"
                  >
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-org-admins-conditional"
                        id="group-org-admins"
                        name="group"
                        type="radio"
                        value="org-admins"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-admins"
                      >
                        Admin
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-org-publishers-conditional"
                        id="group-org-publishers"
                        name="group"
                        type="radio"
                        value="org-publishers"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-publishers"
                      >
                        Publisher
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-org-staff-conditional"
                        id="group-org-staff"
                        name="group"
                        type="radio"
                        value="org-staff"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-staff"
                      >
                        Staff
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-operators-conditional"
                        id="group-operators"
                        name="group"
                        type="radio"
                        value="operators"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-operators"
                      >
                        Operator
                      </label>
                    </div>
                  </div>
                </div>
              </fieldset>
            </div>
            <button
              class="govuk-button mt-8 mr-5"
              data-module="govuk-button"
            >
              Send invitation
            </button>
            <a
              class="govuk-button mt-8 govuk-button--secondary"
              href="/admin/user-management"
              role="button"
            >
              Cancel
            </a>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/changelog"
              >
                Service changelog
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`AddUser > should render correctly with inputs 1`] = `
<DocumentFragment>
  <div
    class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      class="govuk-phase-banner__content"
    >
      <strong
        class="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        class="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and 
        <a
          class="govuk-link"
          href="/feedback"
          id="feedback-link"
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>
  <div
    class="govuk-width-container"
  >
    <button
      class="govuk-back-link mb-7"
      type="button"
    >
      Back
    </button>
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/admin/add-user"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h1
              class="govuk-heading-xl"
            >
              Add new user
            </h1>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="given-name"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="given-name-input"
                >
                  First name
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="given-name-input"
                    maxlength="100"
                    name="givenName"
                    type="text"
                    value="dummy"
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="family-name"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="family-name-input"
                >
                  Last name
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="family-name-input"
                    maxlength="100"
                    name="familyName"
                    type="text"
                    value="user"
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="email"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="email-input"
                >
                  Email address
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w"
                    id="email-input"
                    maxlength="100"
                    name="email"
                    type="text"
                    value="dummy.user@gmail.com"
                  />
                </div>
              </div>
            </div>
            <table
              class="govuk-table"
            >
              <thead
                class="govuk-table__head"
              >
                <tr
                  class="govuk-table__row"
                />
              </thead>
              <tbody
                class="govuk-table__body"
              >
                <tr
                  class="govuk-table__row"
                >
                  <th
                    class="govuk-table__header align-middle"
                    scope="row"
                  >
                    Organisation
                  </th>
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                  <td
                    class="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                  />
                </tr>
              </tbody>
            </table>
            <div
              class="govuk-form-group"
            >
              <fieldset
                class="govuk-fieldset"
                id="group"
              >
                <legend
                  class="govuk-fieldset__legend"
                >
                  <span
                    class="govuk-heading-s govuk-!-margin-bottom-0"
                  >
                    What account do they require?
                  </span>
                </legend>
                <div
                  class=""
                >
                  <div
                    class="govuk-radios"
                    data-module="govuk-radios"
                  >
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        checked=""
                        class="govuk-radios__input"
                        data-aria-controls="group-org-admins-conditional"
                        id="group-org-admins"
                        name="group"
                        type="radio"
                        value="org-admins"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-admins"
                      >
                        Admin
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-org-publishers-conditional"
                        id="group-org-publishers"
                        name="group"
                        type="radio"
                        value="org-publishers"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-publishers"
                      >
                        Publisher
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item govuk-!-margin-bottom-1"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-org-staff-conditional"
                        id="group-org-staff"
                        name="group"
                        type="radio"
                        value="org-staff"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-org-staff"
                      >
                        Staff
                      </label>
                    </div>
                    <div
                      class="govuk-radios__item"
                    >
                      <input
                        class="govuk-radios__input"
                        data-aria-controls="group-operators-conditional"
                        id="group-operators"
                        name="group"
                        type="radio"
                        value="operators"
                      />
                      <label
                        class="govuk-label govuk-radios__label"
                        for="group-operators"
                      >
                        Operator
                      </label>
                    </div>
                  </div>
                </div>
              </fieldset>
            </div>
            <button
              class="govuk-button mt-8 mr-5"
              data-module="govuk-button"
            >
              Send invitation
            </button>
            <a
              class="govuk-button mt-8 govuk-button--secondary"
              href="/admin/user-management"
              role="button"
            >
              Cancel
            </a>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/changelog"
              >
                Service changelog
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
